### ISSUE TITLE
[Loss] Tweedie DistributionLoss not working

### ISSUE THREAD

**Issue by Antoine-Schwartz** [Nixtla/neuralforecast#1159]
### What happened + What you expected to happen

I get an error during the fit with the Tweedie DistributionLoss.
It seems that at the first step, 1) the loss is NaN, and 2) the output doesn't respect the correct typing.

### Versions / Dependencies

Databricks Runtime : [14.3 LTS ML](https://docs.databricks.com/en/release-notes/runtime/14.3lts-ml.html)
+ pip install neuralforecast==1.7.5

### Reproduction script

```
from neuralforecast import NeuralForecast
from neuralforecast.models import NHITS
from neuralforecast.losses.pytorch import DistributionLoss
from neuralforecast.utils import AirPassengersPanel

Y_df = AirPassengersPanel

nf = NeuralForecast(
    models=[
        NHITS(
            h=12,
            input_size=48,
            max_steps=100,
            scaler_type=None,
            loss=DistributionLoss('Tweedie', rho=1.5, validate_args=False),
            enable_model_summary=False,
            enable_checkpointing=False,
            logger=False
        )
    ],
    freq="M"
).fit(Y_df)
```

### Issue Severity

Medium: It is a significant difficulty but I can work around it.

**Comment by elephaint** [comment]
Thanks for using NF!

I've PR'd a fix for this issue (#1164)

**Comment by andmadeira** [comment]
Hi!

I noticed that `log_mu` is clamped twice in `tweedie_scale_decouple` function, both before and after adding `torch.log(loc)`. Could you clarify the reasoning behind the double clamping? It seems that clamping before the adjustment might limit `log_mu` too early, potentially restricting its range before the final scaling step.

Would it be possible to achieve the same numerical stability by applying the clamping only at the final step? This way, `log_mu` is fully adjusted before being constrained, ensuring it remains in a valid range without prematurely truncating values. Let me know your thoughts!

**Comment by elephaint** [comment]
The log range is limited to 1e-9 and 37. That means the range is limited to exp(1e-9), exp(37). So, it doesn't matter at all.

I remember adding it twice because of a failure but I guess we could rid of the first one.
